backend/app/core/search_cache_wrapper.py

```diff
@@ -0,0 +1,69 @@
+import dataclasses
+from typing import Optional
+from fastapi import BackgroundTasks
+
+from app.core.search import OpenSearchConnection, OpenSearchQueryConfig
+from app.api.api_v1.schemas.search import SearchRequestBody, SearchResponseBody
+
+
+class SearchCacheWrapper:
+    """Wrapper for search to handle caching"""
+
+    def __init__(self, os_connection: OpenSearchConnection) -> None:
+        self.os_connection = os_connection
+
+    def is_first_page(self, r: SearchRequestBody):
+        """Is the first page requested for a search."""
+        return r.offset == 0
+
+    @staticmethod
+    def _q(
+        os_connection: OpenSearchConnection,
+        search_request_body: SearchRequestBody,
+        config: OpenSearchQueryConfig,
+        preference: Optional[str],
+    ):
+        """Static function has been created so it can be used as a BackgroundTask."""
+        return os_connection.query(search_request_body, config, preference)
+
+    def query(
+        self,
+        search_request_body: SearchRequestBody,
+        opensearch_internal_config: OpenSearchQueryConfig,
+        preference: Optional[str],
+        background_tasks: Optional[BackgroundTasks] = None,
+    ) -> SearchResponseBody:
+        """Caching equivalent of the OpenSearchConnection query function"""
+
+        if background_tasks is not None and self.is_first_page(search_request_body):
+            # override certain config values to return ASAP
+
+            config = opensearch_internal_config
+
+            overrides = {
+                "max_doc_count": 20,
+                "max_passages_per_doc": 5,
+                "n_passages_to_sample_per_shard": 100_000,
+            }
+            config = dataclasses.replace(opensearch_internal_config, **overrides)
```

Comment:
This will work fine for now, but it might cause some unexpected behaviour for future API use. Nothing to change now (because we're not making the API public at the moment) but maybe something to note in the code for future developers?

Reply:
We could simply create a new endpoint - one which calls the cache wrapper and one that doesn't - naming them appropriately

Reply:
resolving after adding dev agenda item to discuss